packages/lockfile-file/src/experiments/inlineSpecifiersLockfileConverters.ts

```diff
@@ -0,0 +1,105 @@
+import type { Lockfile, ProjectSnapshot, ResolvedDependencies } from '@pnpm/lockfile-types'
+import type { InlineSpecifiersLockfile, InlineSpecifiersProjectSnapshot, InlineSpecifiersResolvedDependencies } from './InlineSpecifiersLockfile'
+
+export function isExperimentalInlineSpecifiersFormat (
+  lockfile: InlineSpecifiersLockfile | Lockfile
+): lockfile is InlineSpecifiersLockfile {
+  return (lockfile as InlineSpecifiersLockfile).inlineSpecifiersFormat ?? false
+}
+
+export function convertToInlineSpecifiersFormat (lockfile: Lockfile): InlineSpecifiersLockfile {
+  return {
+    inlineSpecifiersFormat: true,
+    ...lockfile,
+    importers: mapValues(lockfile.importers, convertProjectSnapshotToInlineSpecifiersFormat),
+  }
+}
+
+export function revertFromInlineSpecifiersFormatIfNecessary (lockfile: Lockfile | InlineSpecifiersLockfile): Lockfile {
+  return isExperimentalInlineSpecifiersFormat(lockfile)
+    ? revertFromInlineSpecifiersFormat(lockfile)
+    : lockfile
+}
+
+export function revertFromInlineSpecifiersFormat (lockfile: InlineSpecifiersLockfile): Lockfile {
+  const { importers, inlineSpecifiersFormat, ...rest } = lockfile
+
+  return {
+    ...rest,
+    importers: mapValues(importers, revertProjectSnapshot),
+  }
+}
+
+function convertProjectSnapshotToInlineSpecifiersFormat (
+  projectSnapshot: ProjectSnapshot
+): InlineSpecifiersProjectSnapshot {
+  const { specifiers } = projectSnapshot
+  return {
```

Comment:
we might have other fields, not only dependenciesMeta, so better do

```suggestion
  return {
    ...projectSnapshot,
```

Reply:
You're right, thanks for catching.

Reply:
Should be fixed in latest push.